### DISCUSSION TITLE
How to listen to the closing event of a webview editor tab

### DISCUSSION THREAD

**Discussion by KrabsWong** [microsoft/vscode-discussions#353] [Extension Development QnA] [answered]
As the title says, is there any API can listen to the closing event?

The condition is I create something with webview and open it in vscode client, then I wanna do something if user close that tab (such as reset some variables and blabla.)

I found an api named `onDidCloseTextDocument`, but it doesn't be fired if I close the opened webview tab.

```typescript
  vscode.workspace.onDidCloseTextDocument((listener) => {
   // not fired
    console.log('onDidCloseTextDocument listener:', listener);
  });
```

**Comment by ArturoDent** [comment, marked as answer]
I haven't used webViews but it looks like `onDidDispose()` or `onDidChangeViewState()` of `WebViewPanel` might work for you.

**Reply by KrabsWong** [reply]
Thanks, `onDidDispose` fits my business scenario perfectly.
